modules/internal/ChapelTuple.chpl

```diff
@@ -97,6 +97,14 @@ module ChapelTuple {
     // body inserted during generic instantiation
   }
 
+  pragma "no doc"
+  proc *(param p: uint, type t) type {
+    if p > max(int) then
+      compilerError("Tuples of size >" + max(int) + " are not currently supported");
+    param pAsInt: int = p: int;
```

Comment:
Is there a reason that the `:int` appears twice here? Also I would have thought that we'd allow coercions from a `param uint` storing say 100 to a `param int` argument. But maybe these are only allowed for the corresponding non-param arguments?

Reply:
> Is there a reason that the :int appears twice here?

I was debugging another puzzle and inserted the second when I started getting superstitious.  I'll remove one.

> I would have thought that we'd allow coercions from a param uint storing say 100 to a param int argument. But maybe these are only allowed for the corresponding non-param arguments?

The problem seems to be that since we have both `param int` and non-`param int` overloads, we get an ambiguity even though the non-param case is flagged as "last resort".  Here's a smaller reproducer:

```chapel
pragma "last resort"
proc takesInt(x: int) {
  writeln("x is ", x);
}

proc takesInt(param x: int) {
  writeln("x is ", x);
}

param ui = 1: uint;
//var t: ui*int;                                                                
writeln(ui.type:string);
takesInt(ui);
```

Also interesting to note is that the output of compiling this is:

```
testit.chpl:21: error: unresolved call 'takesInt(1)'
```

where I thought we'd started labeling literals with types in error messages now...

It might be worth forking off issues for some of these things.  In any case, though, providing the explicit overload has the advantage of generating an error message for uint's > max(int) (where otherwise, we'd get a silent truncation in the compiler).  Granted, this is unlikely to be something that occurs in practice, but...

Reply:
Right, this PR is fine, but I'd appreciate issues for those other problems.

Reply:
I should never fix "simple" bugs... they always turn into hydras...  :P

Reply:
That's what the Icebox is for, right?

Reply:
Keeping Hydras?

Reply:
It's not the existence of the issues that frustrates me, it's the time it requires to create them.

Reply:
I added futures to this PR and opened issues #11705 and #11706 to capture the two issues discussed here.  With more exploration, I'm not sure the "last resort" pragma above had anything to do with anything.  I think we just don't consider param uints as ever dispatching to ints (I thought I'd seen a case where we did earlier, but can't reproduce now).